tests/ovm_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	// "github.com/onsi/ginkgo/extensions/table"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"time"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("OfflineVirtualMachine", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+	})
+
+	Context("A valid OfflineVirtualMachine given", func() {
+
+		newOfflineVirtualMachine := func() *v1.OfflineVirtualMachine {
+			template := tests.NewRandomVMWithEphemeralDisk("kubevirt/cirros-registry-disk-demo:devel")
+			newOVM := NewRandomOfflineVirtualMachine(template, true)
+			newOVM, err = virtClient.OfflineVirtualMachine(tests.NamespaceTestDefault).Create(newOVM)
+			Expect(err).ToNot(HaveOccurred())
+			return newOVM
+		}
+
+		It("should update readyReplicas once VMs are up", func() {
+			newOVM := newOfflineVirtualMachine()
+			Eventually(func() v1.OfflineVirtualMachineConditionType {
+				ovm, err := virtClient.OfflineVirtualMachine(tests.NamespaceTestDefault).Get(newOVM.ObjectMeta.Name, &v12.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return ovm.Status.Conditions[0].Type
+			}, 120*time.Second, 1*time.Second).Should(Equal("Running"))
+		})
+
+		It("should remove VM once the OVM is marked for deletion", func() {
+			newOVM := newOfflineVirtualMachine()
+			// Create a offlinevm with vm
+			// Delete it
+			Expect(virtClient.OfflineVirtualMachine(newOVM.ObjectMeta.Namespace).Delete(newOVM.ObjectMeta.Name, &v12.DeleteOptions{})).To(Succeed())
```

Comment:
You can directly access `Namespace` and `Name` on the object. That makes the code more readable. We still have this pattern a lot in the code because it did not work in the past because of some client-go workarounds.